### PR TITLE
Update README: Update automatic check descriptions

### PR DESCRIPTION
**Pull request by Pandapip1**
They're over half a year out-of-date.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,16 +23,20 @@ Please consider anything which is not published on <https://eips.ethereum.org/>
...
 
-This repository contains an "auto-merge" feature to ease the workload for EIP editors. Pull requests to any EIP will be auto-merged if the EIP's authors approve the PR on GitHub. This is handled by [EIP-Bot](https://github.com/ethereum/EIP-Bot).
+All pull requests in this repository must pass automated checks before they can be automatically merged:
 
-## Validation
+- [EIP-Bot](https://github.com/ethereum/EIP-Bot/) determines when PRs can be automatically merged [^1]
+- EIP-1 rules are enforced using [EIP Walidator](https://github.com/ethereum/eipw)[^2][^3]
+- HTML formatting and broken links are enforced using [HTMLProofer](https://github.com/gjtorikian/html-proofer)[^2]
+- Spelling is enforced with [CodeSpell](https://github.com/codespell-project/codespell)[^2]
+  - False positives sometimes occur. When this happens, please immediately submit a PR editing [.codespell-whitelist](https://github.com/ethereum/EIPs/blob/master/config/.codespell-whitelist).
```
Comment (lightclient) [inline README.md:34]:
```suggestion
- False positives sometimes occur. When this happens, please immediately submit a PR editing [.codespell-whitelist](https://github.com/ethereum/EIPs/blob/master/config/.codespell-whitelist).
```

File: README.md [hunk 2]
```diff
@@ -23,16 +23,20 @@ Please consider anything which is not published on <https://eips.ethereum.org/>
...
+- Markdown best practices are checked using [markdownlint](https://github.com/DavidAnson/markdownlint)[^2]
 
-Pull requests in this repository must pass automated validation checks:
-
-- HTML formatting, broken links, and EIP front matter/formatting are [checked](https://github.com/ethereum/EIPs/blob/master/.github/workflows/ci.yml) using [html-proofer](https://rubygems.org/gems/html-proofer) and [`eipw`](https://github.com/ethereum/eipw).
-- Required pull request reviews are [enforced](https://github.com/ethereum/EIPs/blob/master/.github/workflows/auto-review-bot.yml) using [EIP-Bot](https://github.com/ethereum/EIP-Bot/).
+[^1]: https://github.com/ethereum/EIPs/blob/master/.github/workflows/auto-review-bot.yml
+[^2]: https://github.com/ethereum/EIPs/blob/master/.github/workflows/ci.yml
+[^3]: Not a typo 😈
```
Comment (lightclient) [inline README.md:39]:
```suggestion
```

File: README.md [hunk 3]
```diff
@@ -23,16 +23,20 @@ Please consider anything which is not published on <https://eips.ethereum.org/>
...
 
-## Automerger
+## Validation and Automerging
 
-This repository contains an "auto-merge" feature to ease the workload for EIP editors. Pull requests to any EIP will be auto-merged if the EIP's authors approve the PR on GitHub. This is handled by [EIP-Bot](https://github.com/ethereum/EIP-Bot).
+All pull requests in this repository must pass automated checks before they can be automatically merged:
 
-## Validation
+- [EIP-Bot](https://github.com/ethereum/EIP-Bot/) determines when PRs can be automatically merged [^1]
+- EIP-1 rules are enforced using [EIP Walidator](https://github.com/ethereum/eipw)[^2][^3]
```
Comment (lightclient) [inline README.md:31]:
```suggestion
- EIP-1 rules are enforced using [EIP Walidator](https://github.com/ethereum/eipw)[^2]
```

**Review by lightclient — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (SamWilsn) [inline README.md:31]:
```suggestion
- EIP-1 rules are enforced using [`eipw`](https://github.com/ethereum/eipw)[^2]
```

Name of the tool is `eipw`, not "EIP Walidator".

File: README.md [hunk 4]
```diff
@@ -23,16 +23,19 @@ Please consider anything which is not published on <https://eips.ethereum.org/>
...
 
-This repository contains an "auto-merge" feature to ease the workload for EIP editors. Pull requests to any EIP will be auto-merged if the EIP's authors approve the PR on GitHub. This is handled by [EIP-Bot](https://github.com/ethereum/EIP-Bot).
+All pull requests in this repository must pass automated checks before they can be automatically merged:
 
-## Validation
+- [EIP-Bot](https://github.com/ethereum/EIP-Bot/) determines when PRs can be automatically merged [^1]
+- EIP-1 rules are enforced using [`eipw`](https://github.com/ethereum/eipw)[^2]
+- HTML formatting and broken links are enforced using [HTMLProofer](https://github.com/gjtorikian/html-proofer)[^2]
+- Spelling is enforced with [CodeSpell](https://github.com/codespell-project/codespell)[^2]
+- False positives sometimes occur. When this happens, please immediately submit a PR editing [.codespell-whitelist](https://github.com/ethereum/EIPs/blob/master/config/.codespell-whitelist).
```
Comment (lightclient) [inline README.md:34]:
```suggestion
- False positives sometimes occur. When this happens, please submit a PR editing [.codespell-whitelist](https://github.com/ethereum/EIPs/blob/master/config/.codespell-whitelist).
```

**Review by lightclient — APPROVED** [review]

**Pull request merged**